lib/stdlib/src/lists.erl

```diff
@@ -957,6 +957,24 @@ keymap(Fun, Index, [Tup|Tail]) ->
 keymap(Fun, Index, []) when is_integer(Index), Index >= 1, 
                             is_function(Fun, 1) -> [].
 
+-spec enumerate(List1) -> List2 when
+      List1 :: [T],
+      List2 :: [{Index, T}],
+      Index :: integer(),
+      T :: term().
+enumerate(List1) ->
+    enumerate(List1, 0).
+
+-spec enumerate(List1, Index) -> List2 when
```

Comment:
Should it be `Index, List1` instead perhaps? The list tends to be after other arguments in the rest of the module.

Reply:
Yeah, this is also more consistent with the result type